protocol/x/vault/keeper/orders_test.go

```diff
@@ -218,127 +286,464 @@ func TestRefreshAllVaultOrders(t *testing.T) {
+							ctx,
+							tApp.App,
+							testapp.MustMakeCheckTxOptions{
+								AccAddressForSigning: constants.AliceAccAddress.String(),
+								Gas:                  constants.TestGasLimit,
+								FeeAmt:               constants.TestFeeCoins_5Cents,
+							},
+							&msgRetrieveFromVault,
+						)
+						checkTxResp := tApp.CheckTx(CheckTx_MsgRetrieveFromVault)
+						require.Conditionf(t, checkTxResp.IsOK, "Expected CheckTx to succeed. Response: %+v", checkTxResp)
+
+						return tApp.AdvanceToBlock(
+							uint32(tApp.GetBlockHeight())+1,
+							testapp.AdvanceToBlockOptions{
+								BlockTime: ctx.BlockTime().Add(time.Second * 2),
+							},
+						)
 					},
-					&msgRetrieveFromVault,
-				)
-				checkTxResp := tApp.CheckTx(CheckTx_MsgRetrieveFromVault)
-				require.Conditionf(t, checkTxResp.IsOK, "Expected CheckTx to succeed. Response: %+v", checkTxResp)
+					ordersRefreshed: []bool{
+						true,
+						true,
+						true,
+						true,
+					},
+					orderSides: []clobtypes.Order_Side{
+						clobtypes.Order_SIDE_SELL,
+						clobtypes.Order_SIDE_BUY,
+						clobtypes.Order_SIDE_SELL,
+						clobtypes.Order_SIDE_BUY,
+					},
+					clientIdIsCanonical: []bool{
+						false,
+						false,
+						false,
+						false,
+					},
+				},
+			},
+		},
+		"Success - Orders refresh due to status changing to close-only. No more orders": {
+			vaultId: constants.Vault_Clob0,
+			instances: []Instance{
+				{
+					advanceBlock: func(ctx sdk.Context, tApp *testapp.TestApp) sdk.Context {
+						msgSetVaultParams := vaulttypes.MsgSetVaultParams{
+							Authority: constants.AliceAccAddress.String(), // operator
+							VaultId:   constants.Vault_Clob0,
+							VaultParams: vaulttypes.VaultParams{
+								Status: vaulttypes.VaultStatus_VAULT_STATUS_CLOSE_ONLY,
+							},
+						}
+						CheckTx_MsgSetVaultParams := testapp.MustMakeCheckTx(
+							ctx,
+							tApp.App,
+							testapp.MustMakeCheckTxOptions{
+								AccAddressForSigning: constants.AliceAccAddress.String(),
+								Gas:                  constants.TestGasLimit,
+								FeeAmt:               constants.TestFeeCoins_5Cents,
+							},
+							&msgSetVaultParams,
+						)
+						checkTxResp := tApp.CheckTx(CheckTx_MsgSetVaultParams)
+						require.Conditionf(t, checkTxResp.IsOK, "Expected CheckTx to succeed. Response: %+v", checkTxResp)
 
-				return tApp.AdvanceToBlock(
-					uint32(tApp.GetBlockHeight())+1,
-					testapp.AdvanceToBlockOptions{
-						BlockTime: ctx.BlockTime().Add(time.Second * 2),
+						return tApp.AdvanceToBlock(
+							uint32(tApp.GetBlockHeight())+1,
+							testapp.AdvanceToBlockOptions{
+								BlockTime: ctx.BlockTime().Add(time.Second * 2),
+							},
+						)
 					},
-				)
+					ordersRefreshed:     []bool{}, // no orders
+					orderSides:          []clobtypes.Order_Side{},
+					clientIdIsCanonical: []bool{},
+				},
+			},
+		},
+		"Success - Orders refresh due to status changing to close-only. Sell orders only": {
```

Comment:
nit: Another similar test-case for `STAND_BY` and `DEACTIVATED`?

Reply:
right now we simply skip `refresh orders` if status is `stand_by` or `deactivated`. I'm planning to have another change that cancels existing orders if status is changed to `stand_by` or `deactivated` and will add these test cases in that change